datastore/db/queries/inspection/__init__.py

```diff
@@ -382,3 +386,44 @@ def update_inspection(
         raise InspectionUpdateError(f"Invalid input: {str(e)}") from e
     except Exception as e:
         raise InspectionUpdateError(f"Unexpected error: {str(e)}") from e
+
+
+def delete_inspection(
+    cursor: Cursor,
+    inspection_id: str | UUID,
+    user_id: str | UUID,
+) -> DBInspection:
+    """
+    Delete an inspection from the database and return the deleted inspection record.
```

Comment:
I dont see the value of returning a deleted record. If it is deleted, the data shouldn't be passed around anymore

Reply:
Why: https://github.com/ai-cfia/ailab-datastore/pull/128#discussion_r1741263788

It's fine keeping deleted data in app memory as long as it is known to be so. Because it could contain data still useful elsewhere.

Reply:
you can resolve once read